js/src/admin/utils/ExtensionData.js

```diff
@@ -0,0 +1,34 @@
+import ItemList from '../../common/utils/ItemList';
+import listItems from '../../common/helpers/listItems';
+
+export default class ExtensionData {
+  constructor(extension) {
+    this.extension = extension;
+    app.extensionData[extension] = app.extensionData[extension] || {};
+  }
+
+  registerData(type, content, permissionType = null, priority = 0) {
```

Comment:
IMO, this would be better if split into a `registerPermission` and `registerSetting` methods.